### DISCUSSION TITLE
Importing purchase and project data

### DISCUSSION THREAD

**Discussion by wvolz** [inventree/InvenTree#5414] [Q&A]
What's the recommended way to import purchase and project data for parts? I've been tracking this on a per part basis in a spreadsheet but other than the "notes" field on stock items I don't see where this could live without a PO and project tag. Is that accurate?

Example from a Mouser order:

```
Part, Project, Supplier, Order#, PurchaseDate
71-CPF2-F-301, ProjectX, Mouser, 12345, 01/01/2018
```

**Comment by SchrodingersGat** [comment]
> What's the recommended way to import purchase and project data for parts?

What precisely do you mean by "purchase" and "project" data here?

**Reply by wvolz** [reply]
Good question. For each part I track the following supplier order information/data and the project that required it. 

project data:

- supplier name
- order number / invoice number
- cost
- purchase date

project data:
- name of project that required the part

For me, projects are tasks such as rebuilding a ham radio amplifier, repairing an appliance, building an electronics kit, etc. Additionally, supplier orders can contain parts purchased for multiple projects.

**Reply by SchrodingersGat** [reply]
Ok I think that the best approach is to make a historical purchase order (in InvenTree) which corresponds with each order in your old system. If you wish to track your old orders here, that's the way to do it.

- Supplier name: Link the PO to the particular supplier (you'll have to create the supplier first!)
- Order number / invoice number: Use the PO `reference` field here
- Cost: Add the individual part(s) as line items with the appropriate cost values
- purchase date: This is a field on the PO model
- project name: You can use the "project code" field for this purpose

**Reply by wvolz** [reply]
Ok, I think that makes sense except that project codes can only be assigned at the purchase order level, not line item. My existing "purchase orders" have line items (parts) that belong to multiple projects. Perhaps the "reference" field in the line item can be used for this purpose?

**Reply by wvolz** [reply]
Also, is there a plan to add project codes to purchase order line items?

**Reply by SchrodingersGat** [reply]
@wvolz no "plan" but I'm not against the idea. If you raise an issue for this (or even better, submit a PR to implement it) I'd support that
